User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs

```diff
@@ -0,0 +1,388 @@
+﻿using System;
+using System.Linq;
+using BusinessLayer;
+using BusinessLayer.Reader.Fakes;
+using BusinessLayer.Utilities.Parser.Interfaces.Fakes;
+using BusinessLayer.Utilities.Validator.Interfaces.Fakes;
+using BusinessLayer.Writer.Fakes;
+using Helper;
+using Microsoft.QualityTools.Testing.Fakes;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SortinfApp.UnitTests
+{
+    /// <summary>
+    /// Represents tests for CollectionSortHandler class.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class CollectionSortHandlerTest
+    {
+        /// <summary>
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullParserInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var reader = new StubDataReader(new StubIValidator());
```

Comment:
We need to separate Stubs and Shims.
Stub is used for abstractions only (abstract class and Interface)
Shim is used for instances.

Reply:
Also don't what parameter you pass as null.
This should be specified in Arrange block.